packages/libs/src/sdk/services/Solana/programs/RewardManager/types.ts

```diff
@@ -0,0 +1,71 @@
+import type { PublicKey } from '@solana/web3.js'
+import type { SolanaProgramConfigInternal } from '../types'
+import { z } from 'zod'
+import { PublicKeySchema } from '../../types'
+
+export type RewardManagerConfigInternal = {
+  programId: PublicKey
+  rewardManagerState: PublicKey
+} & SolanaProgramConfigInternal
+
+export type RewardManagerConfig = Partial<RewardManagerConfigInternal>
+
+export const CreateSenderInstructionSchema = z.object({
+  manager: PublicKeySchema,
+  sender: z.string(),
+  operator: z.string(),
+  feePayer: PublicKeySchema.optional()
+})
+
+export type CreateSenderInstructionRequest = z.input<
```

Comment:
just curious why `z.input` rather than `z.infer` here? aren't both just the shape of the above object in this case, or am i missing something?

Reply:
resolved offline - `z.input` has different types as the schema can convert types